ir/instr.h

```diff
@@ -606,13 +716,20 @@ class Memcmp final : public Instr {
 };
 
 
-class Strlen final : public Instr {
+class Strlen final : public MemInstr {
   Value *ptr;
 public:
   Strlen(Type &type, std::string &&name, Value &ptr)
-    : Instr(type, std::move(name)), ptr(&ptr) {}
+    : MemInstr(type, std::move(name)), ptr(&ptr) {}
 
   Value *getPointer() const { return ptr; }
+
```

Comment:
you missed to outline the methods below :)

Reply:
Sorry, it's done.